File: ObjC/SIALogger/SIALogger/Sources/LoggerOutput/SIALogConsoleOutput.m
//
//  SIALogConsoleOutput.m
//  SIALogger
//
//  Created by Ivlev  Alexander (Stef) on 4/15/15.
//  Copyright (c) 2016 Ivlev Alexander. All rights reserved.
//

#import "SIALogConsoleOutput.h"

@implementation SIALogConsoleOutput

- (void)logWithTime:(NSString*)time Level:(SIALogLevel*)level File:(NSString*)file Line:(NSNumber*)line Msg:(NSString*)msg {
  assert(nil != time && nil != level && nil != file && nil != line && nil != msg);
  
  //for faster
  printf("%s [%s] {%s:%lld} %s\r\n", [time UTF8String], [level.name.uppercaseString UTF8String], [file UTF8String], [line unsignedLongLongValue], [msg UTF8String]);
}

@end

File: Swift/SIALogger/SIALogger/LoggerOutput/SIALogConsoleOutput.swift
//
//  SIALogConsoleOutput.swift
//  SIALogger
//
//  Created by Alexander Ivlev on 31/05/16.
//  Copyright © 2016 Alexander Ivlev. All rights reserved.
//

import Foundation

public class SIALogConsoleOutput : SIALogOutputProtocol {
  public static let defaultLogFormat = "%t [%U] {%f:%l}: %m"
  
  public convenience init() {
    self.init(logFormat: SIALogConsoleOutput.defaultLogFormat)
  }
  
  public required init(logFormat: String) {
    formatter = SIALogFormatter(format: logFormat)
  }
  
  public func log(msg: SIALogMessage) {
    print(formatter.toString(msg))
  }
  
  private let formatter: SIALogFormatter
}